src/main/java/org/reaktivity/nukleus/http2/internal/routable/Correlation.java

```diff
@@ -19,25 +19,29 @@
 
 import java.util.Objects;
 
+import org.reaktivity.nukleus.http2.internal.routable.stream.SourceInputStreamFactory;
 import org.reaktivity.nukleus.http2.internal.router.RouteKind;
 
 public class Correlation
 {
     private final String source;
+    private final SourceInputStreamFactory.SourceInputStream sourceInputStream;
```

Comment:
We may need to create a Http2Session(which contains  SourceInputStream and TargetOutputStream)  since reading and writing are changing the connection and stream state.